ovs-service/src/main/java/org/dcsa/ovs/controller/OVSRestController.java

```diff
@@ -0,0 +1,75 @@
+package org.dcsa.ovs.controller;
+
+import lombok.RequiredArgsConstructor;
+import org.dcsa.ovs.transferobjects.ServiceScheduleTO;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.ConstraintViolationException;
+import javax.validation.constraints.Size;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+@Validated
+@RestController
+@RequiredArgsConstructor
+public class OVSRestController {
+
+  @GetMapping(path = "/service-schedules")
+  @ResponseStatus(HttpStatus.OK)
+  public List<ServiceScheduleTO> findAll(
+      @Size(max = 5) @RequestParam(required = false) String carrierServiceCode,
+      @Size(max = 5) @RequestParam(required = false) String universalServiceReference,
+      // TODO: use the vesselIMONumber validator from core
+      @Size(max = 7) @RequestParam(required = false) String vesselIMONumber,
+      @Size(max = 35) @RequestParam(required = false) String vesselName,
+      @Size(max = 50) @RequestParam(required = false) String voyageNumber,
+      @RequestParam(required = false) String universalVoyageReference,
+      @Size(max = 5) @RequestParam(value = "UNLocationCode", required = false)
+          String unLocationCode,
+      @Size(max = 5) @RequestParam(required = false) String facilitySMDGCode,
+      @RequestParam(required = false) String startDate,
+      @RequestParam(required = false) String endDate,
+      @RequestParam(required = false) Integer limit,
+      @RequestParam(required = false) String cursor,
+      @RequestParam(value = "API-Version", required = false) String apiVersion) {
+    return Collections.emptyList();
+  }
+
+  // TODO: Move to a handler
+  @ExceptionHandler(ConstraintViolationException.class)
+  public ResponseEntity<RequestFailureTO> badRequest(
+      HttpServletRequest httpServletRequest, ConstraintViolationException cvex) {
+
+    String exceptionMessage = null;
+
+    if (Objects.nonNull(cvex.getConstraintViolations())) {
```

Comment:
Please just use == null. Objects.nonNull is not meant to be used in this way.

Reply:
It is temporary code, for now, to comply with the error format, I will change it when I extract it out later.